### PR TITLE
[MM-28659] Introducing Thanos utility group

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Introducing Thanos utility group.
- Two R53 records created as part of each Thanos deployment. One targeting then normal ingress and the other one the GRPC ingress that will be used by the central querier. 
- Autoscaling, requests and limits set for Thanos Querier.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28659

#### Release Note
```release-note
- Introducing Thanos utility group.
- Two R53 records created as part of each Thanos deployment. One targeting then normal ingress and the other one the GRPC ingress that will be used by the central querier. 
- Autoscaling, requests and limits set for Thanos Querier.
```

### CODE REVIEW

File: model/cluster_utility.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ const (
 	PrometheusCanonicalName = "prometheus"
 	// PrometheusOperatorCanonicalName is the canonical string representation of prometheus operator
 	PrometheusOperatorCanonicalName = "prometheus-operator"
+	// ThanosCanonicalName is the canonical string representation of prometheus
```
Comment (gabrieljackson) [inline model/cluster_utility.go:17]:
```suggestion
	// ThanosCanonicalName is the canonical string representation of thanos
```

File: internal/provisioner/thanos.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+}
+
+func (t *thanos) Migrate() error {
+	return nil
+}
+
+func (t *thanos) NewHelmDeployment() *helmDeployment {
+	privateDomainName, err := t.awsClient.GetPrivateZoneDomainName(t.logger)
+	if err != nil {
+		t.logger.WithError(err).Error("unable to lookup private zone name")
```
Comment (gabrieljackson) [inline internal/provisioner/thanos.go:178]: We store this value even if there is an error. I see in the `CreateOrUpgrade` method we look up the privateDomainName again. Do we need to even look it up here?

Comment (stylianosrigas) [reply internal/provisioner/thanos.go]: @gabrieljackson Thanks you are right on that. We were doing the same thing in Prometheus, which we shouldn't. I updated both Prometheus and Thanos to pass the DNS as part of the NewHelmDeployment function. This way if there is an error we will catch it in the CreateORUpgrade where the privateDomainname is set.

**Review by gabrieljackson — APPROVED** [review]
👍

**Review by cpanato — APPROVED** [review]

**Pull request merged**